crates/sui-graphql-rpc/schema.graphql

```diff
@@ -0,0 +1,840 @@
+
+  lamportVersion: U64
+  gasEffects: GasEffects
+  objectReads: [Object]
+  objectChanges: [ObjectChange]
+  balanceChanges: [BalanceChange]
+
+  epoch: Epoch
+  checkpoint: Checkpoint
+
+  events: TransactionEventConnection
+
+  bcs: Bytes
+}
+
+enum ExecutionStatus {
+  SUCCESS
+  FAILURE
+}
+
+type GasInput {
+  gasSponsor: Address
+  gasPayment: [Object]
+
+  gasPrice: U64
+  gasBudget: U64
+}
+
+type GasEffects {
+  gasObject: Coin
+  gasSummary: GasCostSummary
+}
+
+type GasCostSummary {
+  computationCost: U64
+  storageCost: U64
+  storageRebate: U64
+  nonRefundableStorageFee: U64
+}
+
+type ObjectChange {
+  inputState: Object
+  outputState: Object
+
+  idCreated: Boolean
+  idDeleted: Boolean
+}
+
+type BalanceChange {
+  owner: Owner
+  coinType: MoveType
+  amount: I128
+}
+
+type Event {
+  moduleId: MoveModuleId
+  type: MoveType
+  sender: Address
+  timestampMs: U64
+
+  json: JSON
+  bcs: Bytes
+}
+
+type Balance {
+  coinType: MoveType
+  coinObjectCount: U64
+  totalBalance: U128
+}
+
+type Coin {
+  balance: U64
+  asMoveObject: MoveObject
+}
+
+type StakedSui {
+  status: StakeStatus
+  requestEpoch: Epoch
+  activeEpoch: Epoch
+  principal: U64
+
+  # Only available if status is `ACTIVE`.
+  estimatedReward: U64
+
+  asMoveObject: MoveObject
+}
+
+enum StakeStatus {
+  PENDING
+  ACTIVE
+  UNSTAKED
+}
+
+type CoinMetadata {
+  decimals: Int
+  name: String
+  symbol: String
+  description: String
+  iconURL: String
+  supply: U64
+
+  asMoveObject: MoveObject
+}
+
+input DynamicFieldName {
+  type: TypeInst!
+  bcs: Bytes!
+}
+
+type DynamicField {
+  name: MoveValue
+  value: DynamicFieldValue
+}
+
+union DynamicFieldValue = MoveObject | MoveValue
+
+type MoveObject {
+  contents: MoveValue
+  hasPublicTransfer: Boolean
+
+  asCoin: Coin
+  asStakedSui: StakedSui
+  asObject: Object
+}
+
+type MovePackage {
+  modules: MovePackageModuleConnection
+
+  normalizedModules: NormalizedModulesConnection
+  normalizedModule(name: String!): NormalizedModule
+
+  asObject: Object
+}
+
+type MoveModuleId {
+  package: MovePackage
+  name: String
+}
+
+enum MoveAbility {
+  COPY
+  DROP
+  STORE
+  KEY
+}
+
+enum MoveVisibility {
+  PUBLIC
+  PRIVATE
+  FRIEND
+}
+
+enum MoveReference {
+  IMMUT
+  MUT
+}
+
+type MoveStructTypeParameterDecl {
+  constraints: [MoveAbility]
+  isPhantom: Boolean
+}
+
+type MoveFunctionTypeParameterDecl {
+  constraints: [MoveAbility]
+}
+
+type NormalizedModule {
+  fileFormatVersion: Int
+
+  moduleId: MoveModuleId
+  friends: [MoveModuleId]
+
+  structs: NormalizedStructConnection
+  struct(name: String!): NormalizedStruct
+
+  functions: NormalizedFunctionConnection
+  function(name: String!): NormalizedFunction
+}
+
+type NormalizedStruct {
+  module: NormalizedModule
+  name: String
+
+  abilities: [MoveAbility]
+  typeParameters: [MoveStructTypeParameterDecl]
+  fields: [NormalizedField]
+}
+
+type NormalizedField {
+  name: String
+  type: MoveTypeUse
+}
+
+type NormalizedFunction {
```

Comment:
JSON RPC has a method to get a function's arguments in terms of whether they are pure, object-by-value, object-by-ref, or object-by-mut-ref.  I omitted those, because with the GraphQL API you can extract the necessary information to figure that out with one query.